storage/range_gc_queue.go

```diff
@@ -60,7 +60,7 @@ func (q *rangeGCQueue) needsLeaderLease() bool {
 // and if so at what priority. Currently all inactive ranges are
 // considered for possible GC at equal priority.
 func (q *rangeGCQueue) shouldQueue(now proto.Timestamp, rng *Range) (bool, float64) {
-	if _, expired := rng.HasLeaderLease(now); !expired {
+	if held, expired := rng.HasLeaderLease(now); held && !expired {
```

Comment:
`held` refers to whether *this* node holds the lease; `expired` refers to whether *anyone* does. We want the latter here as explained in the comment. If no one has ever held the lease, expired is true, so we continue to return `true` from `shouldQueue`. In other words, the old code looks correct to me. Was your new test failing without this change?